packages/react/src/components/TextInput/TextInput.js

```diff
@@ -177,6 +257,26 @@ TextInput.propTypes = {
    * `true` to use the light version.
    */
   light: PropTypes.bool,
+
+  /**
+   * Specify if the user should be able to edit the value of the input
+   */
+  readOnly: PropTypes.bool,
+
+  /**
+   * The label for the read-only status icon
+   */
+  readOnlyIconLabel: PropTypes.string,
+
+  /**
+   * Specify whether the character counter is shown
+   */
+  charCount: PropTypes.bool,
+
+  /**
+   * The maximum allowed input value length
+   */
+  maxLength: PropTypes.number,
```

Comment:
Need prop type of `onChange`, too, with "missing `onChange` in controlled mode" check.

Reply:
this should be addressed now